mp/src/game/shared/momentum/mom_gamemovement.cpp

```diff
@@ -2150,16 +2119,18 @@ int CMomentumGameMovement::TryPlayerMove(Vector *pFirstDest, trace_t *pFirstTrac
     float fSlamVol = 0.0f;
 
     float fLateralStoppingAmount = primal_velocity.Length2D() - mv->m_vecVelocity.Length2D();
-    if (fLateralStoppingAmount > PLAYER_MAX_SAFE_FALL_SPEED * 2.0f)
+    if (fLateralStoppingAmount > PLAYER_MAX_SAFE_FALL_SPEED)
     {
-        fSlamVol = 1.0f;
+        fSlamVol = 0.85f;
     }
-    else if (fLateralStoppingAmount > PLAYER_MAX_SAFE_FALL_SPEED)
+    // Increase volume for a harder collision.
+    if (fLateralStoppingAmount > PLAYER_MAX_SAFE_FALL_SPEED * 2.0f)
     {
-        fSlamVol = 0.85f;
+      fSlamVol = 1.0f;
     }
 
-    PlayerRoughLandingEffects(fSlamVol);
+    // Use the last traced surface for the landing sound.
+    PlayerRoughLandingEffects(fSlamVol, pm.surface.surfaceProps);
```

Comment:
Since calling this method only does anything if `fSlamVol` is `> 0.0`, you can go back to the logic you had before, and do:

```
if (fLateralStoppingAmount > PLAYER_MAX_SAFE_FALL_SPEED)
{
   float fSlamVol = (fLateralStoppingAmount > PLAYER_MAX_SAFE_FALL_SPEED * 2.0f) ? 1.0f : 0.85f;  
   PlayerRoughLandingEffects(fSlamVol, pm.surface.surfaceProps);
}
```